File: vortexdbg-api/src/main/java/com/vortexdbg/thread/Task.java
package com.vortexdbg.thread;

import com.vortexdbg.AbstractEmulator;
import com.vortexdbg.Emulator;
import com.vortexdbg.signal.SignalOps;
import com.vortexdbg.signal.SignalTask;

import java.util.List;

public interface Task extends SignalOps, RunnableTask {

    String TASK_KEY = Task.class.getName();

    int getId();

    Number dispatch(AbstractEmulator<?> emulator) throws PopContextException;

    boolean isMainThread();

    boolean isFinish();

    void addSignalTask(SignalTask task);

    List<SignalTask> getSignalTaskList();

    void removeSignalTask(SignalTask task);

    boolean setErrno(Emulator<?> emulator, int errno);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/thread/Task.kt
package com.vortexdbg.thread

import com.vortexdbg.AbstractEmulator
import com.vortexdbg.Emulator
import com.vortexdbg.signal.SignalOps
import com.vortexdbg.signal.SignalTask

interface Task : SignalOps, RunnableTask {

    fun getId(): Int

    @Throws(PopContextException::class)
    fun dispatch(emulator: AbstractEmulator<*>): Number?

    fun isMainThread(): Boolean

    fun isFinish(): Boolean

    fun addSignalTask(task: SignalTask)

    fun getSignalTaskList(): List<SignalTask>

    fun removeSignalTask(task: SignalTask)

    fun setErrno(emulator: Emulator<*>, errno: Int): Boolean

    companion object {
        @JvmField
        val TASK_KEY: String = Task::class.java.name
    }

}
